### DISCUSSION TITLE
godot4 Exporting web functional on macos and Ios

### DISCUSSION THREAD

**Discussion by jl-pi** [godotengine/godot-proposals#8896] [Engine Core]
Hi, do you guys know when this upstream bug will be fixed ? Since currently exporting a project to web is not working correctly on mac chrome. I would love to have an estimation for project planning. Like in 2-3 months ? 6 months or 2 -3 years?
Thank you!
```
Godot 4's HTML5 exports currently cannot run on macOS and iOS due to upstream bugs with SharedArrayBuffer and WebGL 2.0. We recommend using macOS and iOS native export functionality instead, as it will also result in better performance.
```

**Comment by AThousandShips** [comment]
See:
* https://github.com/godotengine/godot/issues/70796

**Comment by Calinou** [comment]
We don't know when the [Chrome on macOS issue](https://github.com/godotengine/godot/issues/70691) will be definitely resolved, as it's reliant on upstream issues that are out of our control (such as Chrome on macOS switching to ANGLE's Metal backend).

That said, [compiling a web export template without threads](https://github.com/godotengine/godot/pull/85939) seems to result in improved loading times already, which may be enough to resolve the issue. This feature is expected to land in [4.3 which aims to be released in April 2024](https://docs.godotengine.org/en/latest/about/release_policy.html#release-support-timeline). This doesn't help for Firefox on macOS according to [this comment](https://github.com/godotengine/godot/issues/70691#issuecomment-1852330289) though.

Closing, as this platform isn't for support questions.
